Lib/test/test_urlparse.py

```diff
@@ -1035,8 +1036,29 @@ def test_telurl_params(self):
         self.assertEqual(p1.path, '863-1234')
         self.assertEqual(p1.params, 'phone-context=+1-914-555')
 
+    def test_clear_cache_deprecation(self):
+        with warnings.catch_warnings(record=True) as w:
+            warnings.simplefilter("always")
+            urllib.parse.clear_cache()
+        self.assertEqual(len(w), 1, msg=repr(w))
+        v = sys.version_info
+        exc = PendingDeprecationWarning if v <= (3, 11) else DeprecationWarning
+        self.assertIs(w[0].category, exc)
+        self.assertIn('clear_cache() will be removed', str(w[0].message))
+
+    def test_Quoter_deprecation(self):
+        with warnings.catch_warnings(record=True) as w:
+            warnings.simplefilter("always")
+            old_class = urllib.parse.Quoter
+            self.assertIs(old_class, urllib.parse._Quoter)
+        self.assertEqual(len(w), 1, msg=repr(w))
+        v = sys.version_info
+        exc = PendingDeprecationWarning if v <= (3, 11) else DeprecationWarning
```

Comment:
I would go directly to a DeprecationWarning.  We haven't had much luck with people ever seeing and benefitting from a PendingDeprecationWarning.

Reply:
I'll defer deciding this until I do some OSS searching to see if I can even find _any_ users. (i bet not, these are obscure)

Reply:
I can see two usages in grepping top 1000 PyPI packages.

```
$ rg -t py 'urllib.parse.*clear_cache'
twisted/src/twisted/web/test/test_http.py
17:from urllib.parse import clear_cache  # type: ignore[attr-defined]

gevent/src/gevent/testing/leakcheck.py
130:            sys.modules['urllib.parse'].clear_cache()
```

Reply:
Thanks!  Based on that (and our own libregrtest use), i'll just keep clear_cache and not deprecate it.